cmd/wharf/root.go

```diff
@@ -129,3 +145,43 @@ func parseLevel(lvl string) (logger.Level, error) {
 		return logger.LevelDebug, fmt.Errorf("invalid logging level: %q", lvl)
 	}
 }
+
+func callParentPersistentPreRuns(cmd *cobra.Command, args []string) error {
+	cmd = cmd.Parent()
+	if cmd == nil {
+		return nil
+	}
+	// Call the root-most one first
+	if err := callParentPersistentPreRuns(cmd, args); err != nil {
+		return err
+	}
+	if cmd.PersistentPreRun != nil {
+		cmd.PersistentPreRun(cmd, args)
+	}
+	if cmd.PersistentPreRunE != nil {
+		return cmd.PersistentPreRunE(cmd, args)
+	}
+	return nil
+}
```

Comment:
There's some bugs in my initial suggestion. It should'nt be recursive, as that would make it call the upper ones multiple times, and it shouldn't call the non-error variant as well, as that one should have its own "callParent" function if we use it.

```suggestion
func callParentPersistentPreRuns(cmd *cobra.Command, args []string) error {
	for {
		cmd = cmd.Parent()
		if cmd == nil {
			return nil
		}
		// Call first one we find.
		// It'll be responsible for calling its parent PersistentPreRunE
		if cmd.PersistentPreRunE != nil {
			return cmd.PersistentPreRunE(cmd, args)
		}
	}
}
```